### PR TITLE
Adding a new scalable IC system for creating large

### PR DESCRIPTION
**Pull request by permcody**
polycrystal systems that are properly colored initially.

This system *should* eventually replace the existing
"reduced" order parameter ICs used elsewhere.

refs #8810

### CODE REVIEW

File: modules/phase_field/src/postprocessors/GrainTracker.C [hunk 1]
```diff
@@ -293,15 +293,15 @@ GrainTracker::finalize()
 
   if (_ebsd_reader && _first_time)
   {
-    expandEBSDGrains();
+    expandPointHalos();
 
     /**
      * By expanding the EBSD Grains we've effectively erased one level of halo.
```
Comment (dschwen) [inline modules/phase_field/src/postprocessors/GrainTracker.C]: might want to get rid of any mention of EBSD specifically

File: modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    _dim(_mesh.dimension()),
+    _op_num(_vars.size()),
+    _grain_num(getParam<unsigned int>("grain_num")),
+    _grain_to_op(_grain_num, PolycrystalUserObjectBase::INVALID_COLOR),
+    _coloring_algorithm(getParam<MooseEnum>("coloring_algorithm")),
+    _initialized(false),
+    _output_adjacency_matrix(getParam<bool>("output_adjacency_matrix"))
+{
+  if (_op_num > _grain_num)
+    mooseError("ERROR in PolycrystalVoronoi: Number of order parameters (op_num) can't be larger "
```
Comment (dschwen) [inline modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: why not?

**Review by dschwen — APPROVED** [review]
Looks fine to me

File: modules/phase_field/src/postprocessors/GrainTracker.C (hunk 1 above)
Comment (permcody) [inline modules/phase_field/src/postprocessors/GrainTracker.C]: That's the goal, this PR won't get us there yet. I plan to modify the existing EBSD IC (ReconVarIC) and eventually remove all traces of this logic from the GrainTracker!

File: modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C (hunk 2 above)
Comment (permcody) [inline modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: This logic was copied over from the existing class. There's actually no reason to limit it other than it's dumb.

**Pull request merged**